great_expectations/data_context/types/base.py

```diff
@@ -1950,12 +1957,26 @@ def validate_schema(self, data, **kwargs):
     # noinspection PyUnusedLocal
     @post_dump
     def remove_keys_if_none(self, data, **kwargs):
-        data = deepcopy(data)
+        data = copy.deepcopy(data)
         for key in self.REMOVE_KEYS_IF_NONE:
             if key in data and data[key] is None:
                 data.pop(key)
         return data
 
+    def dump(self, obj: Any, *, many: Optional[bool] = None) -> dict:
+        batch_data_references: Tuple[
+            Optional[Any], Optional[List[Any]]
+        ] = get_runtime_parameters_batch_data_references_from_config(config=obj)
+        delete_runtime_parameters_batch_data_references_from_config(config=obj)
+        data: dict = super().dump(obj=obj, many=many)
+        restore_runtime_parameters_batch_data_references_into_config(
+            config=obj, batch_data_references=batch_data_references
+        )
+        restore_runtime_parameters_batch_data_references_into_config(
+            config=data, batch_data_references=batch_data_references
+        )
+        return data
+
 
 class CheckpointConfig(BaseYamlConfig):
     # TODO: <Alex>ALEX (does not work yet)</Alex>
```

Comment:
Can we remove this?

Reply:
@NathanFarmer We still need those comments -- the idea was good, but it did not work, and we ought to figure out how to make it work -- or discuss it and conclude that the current approach is the best that can be done.  Thanks!